### PR TITLE
chore(rust): fix a bunch of cargo warnings & errors

### PR DESCRIPTION
**Pull request by universalmind303**
down from ~60 errors to 7 🚀 

We should wait until after #7495 is merged as this is likely to have conflicts.

### CODE REVIEW

File: polars/polars-lazy/polars-plan/src/logical_plan/iterator.rs [hunk 1]
```diff
@@ -321,27 +321,27 @@ impl<'a> Iterator for AlpIter<'a> {
...
 
-    use super::*;
+    // use super::*;
 
     #[test]
     fn test_lp_iter() -> PolarsResult<()> {
-        let df = df! {
-            "a" => [1, 2]
-        }?;
+        // let df = df! {
```
Comment (ritchie46) [inline polars/polars-lazy/polars-plan/src/logical_plan/iterator.rs:331]: This one can be removed.

**Comment by stinodego** [comment]
60 errors? Your rust-analyzer is more generous than mine 😅 
image

This exploded in a recent change, not sure what happened though 🤔 

Thanks for the cleanup!

**Comment by ritchie46** [comment]
@stinodego I think you must activate all features in rust-analyzer's settings.

**Comment by stinodego** [comment]
I had some odd stuff in my Cargo registry, cleaning it out solved my issues. Thanks for the help though!

**Pull request merged**